src/core/load-dropin.c

```diff
@@ -25,10 +25,14 @@ static int process_deps(Unit *u, UnitDependency dependency, const char *dir_suff
                 return r;
 
         STRV_FOREACH(p, paths) {
-                _cleanup_free_ char *target = NULL;
-                const char *entry;
+                _cleanup_free_ char *target = NULL, *target_file = NULL, *entry = NULL;
```

Comment:
Having `entry` be an allocated thing that has to be freed is what causes the test failure with dropins.

Line 86 is now wrong:

```c
r = unit_add_dependency_by_name(u, dependency, entry, true, UNIT_DEPENDENCY_FILE);
```

because that string will be freed when it goes out of scope.  I am not quite sure the best way to handle this.

Reply:
not sure i follow? line 86 is in the same scope as this definition here, so everything should be fine?

Reply:
Now that `entry` is allocated, it is invalid to pass it as an argument to `unit_add_dependency_by_name` because it will be freed at the end of this `process_deps` function since it is declared `_cleanup_free_`.  But if it isn't cleaned up, it will just leak.

Reply:
I don't grok this. `unit_add_dependency_by_name` wouldn't hold the passed name AFAICS.

Reply:
yeah, i don't grok the proble here either? we generally expect functions to not keep references to strings with pass them. THere are very few exceptions, and we usually mark them with explicit comments along the lines of "string is donated here" or similar.